pkg/controllers/podgroup/pg_controller_handler.go

```diff
@@ -61,6 +62,28 @@ func (pg *pgcontroller) addPod(obj interface{}) {
 	}
 
 	pg.queue.Add(req)
+	pg.deletePodGroup(obj)
+}
+
+func (pg *pgcontroller) updatePod(oldObj, newObj interface{}) {
+	pg.deletePodGroup(newObj)
+}
+
+// Delete PodGroup if the Pod is Succeed or Failed.
+func (pg *pgcontroller) deletePodGroup(newObj interface{}) {
+	pod, ok := newObj.(*v1.Pod)
+	if !ok {
+		klog.Errorf("Failed to convert %v to v1.Pod", newObj)
+		return
+	}
+
+	if pod.Status.Phase == v1.PodSucceeded || pod.Status.Phase == v1.PodFailed {
+		pgName := batchv1alpha1.PodgroupNamePrefix + string(pod.UID)
```

Comment:
Better check whether a pod has a controller ownerreference to distinguish rs/sts and noraml pod, and pod has a controller ownerreference  should not go throuth here.

Reply:
Yes, your suggestion is correct. If the OwnerReferences of the pod are not empty, processing will not continue